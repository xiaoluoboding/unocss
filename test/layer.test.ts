import { createGenerator } from '@unocss/core'
import { describe, expect, test } from 'vitest'

describe('layers', () => {
  test('static', async() => {
    const uno = createGenerator({
      rules: [
        ['a', { name: 'bar1' }, { layer: 'a' }],
        ['b', { name: 'bar2' }, { layer: 'b' }],
        [/^c(\d+)$/, ([, d]) => ({ name: d }), { layer: 'c' }],
        [/^d(\d+)$/, ([, d]) => `/* RAW ${d} */`, { layer: 'd' }],
      ],
      shortcuts: [
        ['abcd', 'a b c2 d3', { layer: 's' }],
        ['abc', 'a b c4'],
      ],
      presets: [],
    })
    const { css } = await uno.generate('a b abc abcd d4 c5')
    expect(css).toMatchSnapshot()
  })
})
