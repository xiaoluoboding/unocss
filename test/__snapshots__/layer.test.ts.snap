// Vitest Snapshot v1

exports[`layers > static 1`] = `
"/* layer: shortcuts */
.abc{name:bar1;name:bar2;name:4;}
/* layer: a */
.a{name:bar1;}
/* layer: b */
.b{name:bar2;}
/* layer: c */
.c5{name:5;}
/* layer: d */
/* RAW 4 */
/* layer: s */
.abcd{name:bar1;name:bar2;name:2;}"
`;
